### PR TITLE
CheckoutView API endpoint

### PR DESCRIPTION
**Pull request by mjfrey**
Checkout API endpoint

### CODE REVIEW

**Comment by mjfrey** [comment]
@clintonb @bderusha This is the new PR.  I messed up the old one.  Working on the JS test now.

**Comment by clintonb** [comment]
Given that the JavaScript file is not used, I would be fine if you merged without those files. They still need to be tested; but, once removed, they are no longer blocking this from being merged.

**Comment by mjfrey** [comment]
@clintonb added JS tests.

File: ecommerce/static/js/test/specs/pages/checkout_payment_spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                it('should fill form inputs for each data key/value pair', function () {
+                    var data = {
+                        payment_page_url: 'http://www.dummy-url.com/',
+                        type: 'Dummy Type',
+                        model: '500',
+                        color: 'white',
+                    };
+                    spyOn(_, 'each');
+                    CheckoutPayment.onDone(data);
+                    expect(_.each.calls.count()).toEqual(1);
```
Comment (clintonb) [inline ecommerce/static/js/test/specs/pages/checkout_payment_spec.js]: The behavior we care most about for this function is the form submission. Counting the calls to `_.each()` doesn't validate the functionality we care most about.

File: ecommerce/static/js/test/specs/pages/checkout_payment_spec.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            });
+
+            describe('onReady', function () {
+                it('should disable payment button before making ajax call', function () {
+                    $('<div class="payment-buttons"><button class="payment-button">Pay</button></div>')
+                        .appendTo('body');
+                    spyOn(Utils, 'disableElementWhileRunning');
+                    CheckoutPayment.onReady();
+                    $('button.payment-button').trigger('click');
+                    expect(Utils.disableElementWhileRunning).toHaveBeenCalled();
```
Comment (clintonb) [inline ecommerce/static/js/test/specs/pages/checkout_payment_spec.js]: As above, disabling the button is not what we care most about. We want to make sure the function issues an AJAX request.

**Comment by clintonb** [comment]
These JavaScript tests are not sufficient as they don't actually test checkout initiation. As I mentioned earlier, given that this functionality is not needed by this PR, I recommend removing it. We can review the code again in a fronted PR that actually makes use of the JavaScript.

**Comment by mjfrey** [comment]
Moved JS out of this PR and into the basket PR.

**Comment by clintonb** [comment]
:+1: pending:
- [x] All JS files removed (If the code isn't being used, it doesn't need to be in the PR/repository.)
- [x] Commits squashed into single commit

**Pull request merged**